### ISSUE TITLE
`no-base-to-string` is too eager

### ISSUE THREAD

**Issue by JakobJingleheimer** [oxc-project/tsgolint#517]
It complains about `toString()` on objects that it's perfectly fine to call on, for example `URL`'s `toString()`.

**Comment by camc314** [comment]
Hi, can you please provide a reproduction for this issue?

I tried to reproduce it here: https://github.com/camc314/oxlint-repros/tree/main/issue-tsgolint-517 and it works as expected.

Can you run with `--type-check` to see if there are any type errors?

<img width="522" height="516" alt="Image" src="https://github.com/user-attachments/assets/ca448ac9-1220-4476-87e5-a87ad30e0e56" />

**Comment by JakobJingleheimer** [comment]
```js
mock_undiciRequest.mockImplementationOnce(function(url) {
  if (url.toString() === '…') return Response(…)
});
```

`url`'s type is: `string | URL | UrlObject`. UrlObject is a plain object (rather than an explicit instance of URL) that implements URL's getters, setters, props, and methods. It comes from node's types. Presumably it exists to facilitate testing where the high overhead of an actual url instance is unnecessary.

**Comment by camc314** [comment]
I see, i'm assuming `UrlObject` is from node? if so, this is working as expected, as `toString` isn't defined as a method on `UrlObject`, it's inherting it from the object prototype type definition.

<img width="426" height="115" alt="Image" src="https://github.com/user-attachments/assets/8c9cb0f2-d3c1-4ae2-9538-3899aa9d5a69" />

To fix this, you can either extend node's type definitions like so:

```ts
declare module 'url' {
    interface UrlObject {
        toString(): string;
    }
}
```

This tells typescript that it has a `toString` method that's different than `Object.prototype.toString`.

Unfortunately the `ignoredTypeNames` options isn't supported at present (i'm going to make sure this is shipped by next monday (the next oxlint release.

That should allow you to configure oxlint like:

```json
{
    "rules": {
        "typescript/no-base-to-string": [
            "error",
            {
                "ignoredTypeNames": ["UrlObject"]
            }
        ]
    }
}
```

And it'll ignore the error

**Comment by camc314** [comment]
PR to add the missing options support: https://github.com/oxc-project/oxc/pull/16898

**Comment by JakobJingleheimer** [comment]
Yes, it's from Node.

Ahh. I missed that it was missing in the `UrlObject` type. I guess yes, it's catching a legit problem (even though we never actually use `UrlObject`).

**Comment by camc314** [comment]
All good - thanks for reporting! It surfaceed that we lack the options support for this rule.
